pandas/core/dtypes/cast.py

```diff
@@ -1652,7 +1652,9 @@ def construct_1d_arraylike_from_scalar(
 
     if isinstance(dtype, ExtensionDtype):
         cls = dtype.construct_array_type()
-        subarr = cls._from_sequence([value] * length, dtype=dtype)
+        subarr = cls._from_sequence([value], dtype=dtype)
+        taker = np.broadcast_to(np.intp(0), length)
```

Comment:
can we *just* use `.repeat()`?

Reply:
Yes, that's even better - just updated. Thanks for pointing it out.